### DISCUSSION TITLE
Use react native pan responder with Mapbox.MapView version 10

### DISCUSSION THREAD

**Discussion by chris-gaona** [rnmapbox/maps#2960] [General]
In version 8 & with iOS, I was able to attach a pan responder to the `MapboxGL.MapView` like in the following code & it would allow me to get events from `onPanResponderMove` when the map view was moved/panned. I used this to allow dragging various map shapes....points, line strings, polygons, etc. After upgrading to version 10, this functionality no longer works but I'm suspecting it might be because something hasn't been implemented yet.

Does anyone have any thoughts on how I could implement this functionality? I'd be happy to take a stab at making updates to the swift code if someone could point me in the right direction.

Thanks!

```tsx
<MapboxGL.MapView
  ref={mapRef}
  {...panGestureRef.current.panHandlers}
  ...
></MapboxGL.MapView>
```

**Comment by mfazekas** [comment]
@chris-gaona sorry I have no idea about that. V10 has different gesture handling logic. It might be easier to create new callbacks for this usecase, see gestureManager didBegin, etc.

**Reply by chris-gaona** [reply]
Okay thank you for the reply. I thought I'd ask & see. I was already looking at the gestureManager part in the swift code so it seems I'm on the right track. Thanks!
